### ISSUE TITLE
Generate tokens across all GPUs

### ISSUE THREAD

**Issue by gcooper-isi** [gcooper-isi/DeepSpeedExamples#18]
When model parallelism is used, each GPU (within each group of model parallel GPUs) holds a separate section of the output layer. However, currently, generate_samples.py only samples from the output layer on the first of these GPUs. This task is to fix generation so that it samples across the entire vocabulary, regardless of which GPU holds the corresponding logits.
